src/app/withdraw/components/withdraw-client.tsx

```diff
@@ -230,12 +242,20 @@ const Withdraw = () => {
             isFailed={isError}
             isLoading={isLoading}
             maxFee={maxFee}
-            validationSchema={amountValidationSchema as any}
-            handleSubmit={(value) => {
-              setFieldValue("amount", value);
+            validationSchema={amountValidationSchema}
+            handleSubmit={async (value) => {
+              const limits = await refetch();
+
+              const maxWithdrawalCap =
+                (limits.data?.perWithdrawalCap || 0) / 1e8;
+
+              if (value > maxWithdrawalCap) {
+                return getMaxError(maxWithdrawalCap);
+              }
+              setFieldValue("address", value);
```

Comment:
is `address` instead of `amount` correct here?

Reply:
correct